### PR TITLE
Clean up LogEntry class

### LINKED ISSUE #3878: Clean up LogEntry class
The LogEntry class has some technical debt  that would be good to cleanup.  The following improvements could be made.

 * There is a field named filename that actually seems to store a fully qualified path.  Could rename this field  Could also add validation to this field to ensure its of the expected format.  The filename in path current has a tserver host plush uuid, could validate this.
 * There is a field named extent that is only used by unit test, could probably remove this.
 * There is no equals or hashCode method, these could be added.
 * There is upgrade code in the class that deals with some old format of the data.   Can not see any test for this.   Could remove this if adding validation that would fail on seeing this. Could possibly look for this in upgrade and covert,if we can figure out what the old format was.  However the old format may be really old and seeing it may be unexpected or not realistically possible.
 * The value in the metadata table seems to store the exact same data as the column qualifier.  This code could be cleaned up to not store redundant info, may require change in upgrade.

### PR DESCRIPTION
**Pull request by DomGarguilo**
Closes #3878

From the ticket there are several actionable points:
* rename `filename` to `filepath`
* add validation that `filepath` is a tserver + UUID
* remove `extent` 
* add `equals()` and `hashCode()`
* figure out whether to remove or handle upgrade of old data format
* remove redundant data between metadata table and column qualifier, this might require change in upgrade

So far in this PR:
* renamed `filename` to `filepath`
* removed `extent` 
* added `equals()` and `hashCode()`
* removed `getColumnFamily()` since it was just a wrapper for an existing constant

Left to do:
* implement filePath validation method
* handle old data format
* remove redundant data

### CODE REVIEW

File: core/src/main/java/org/apache/accumulo/core/tabletserver/log/LogEntry.java [hunk 1]
```diff
@@ -19,67 +19,95 @@
...
+
 public class LogEntry {
-  private final KeyExtent extent;
-  public final long timestamp;
-  public final String filename;
 
-  public LogEntry(KeyExtent extent, long timestamp, String filename) {
-    // note the prevEndRow in the extent does not matter, and is not used by LogEntry
-    this.extent = extent;
+  private final long timestamp;
```
Comment (keith-turner) [inline core/src/main/java/org/apache/accumulo/core/tabletserver/log/LogEntry.java:32]: Not something to do in this PR and I am not sure about it, but this timestamp may be able to be removed also.  I need to look into this.

**Review by keith-turner — APPROVED** [review]

**Comment by DomGarguilo** [comment]
Marked this as ready for review. I think I addressed the points from #3878

File: core/src/main/java/org/apache/accumulo/core/tabletserver/log/LogEntry.java (hunk 1 above)
Comment (keith-turner) [reply core/src/main/java/org/apache/accumulo/core/tabletserver/log/LogEntry.java:33]: I'll open another issue about this, but I am fairly certain that the timestamp can be removed.  Its used to order files (but the order seems arbitrary) and when sorted wals are used for recovery a merged sorted view of the files is [constructed](https://github.com/apache/accumulo/blob/a4cc5f82c37b4ff1cb2bb44d99e0b7cef1302fc1/server/tserver/src/main/java/org/apache/accumulo/tserver/log/RecoveryLogsIterator.java#L113) in which order does not matter.  I think the order of the files used to matter, but no longer does and this is vestigial code.

Comment (keith-turner) [reply core/src/main/java/org/apache/accumulo/core/tabletserver/log/LogEntry.java:33]: Opened #3922

File: core/src/main/java/org/apache/accumulo/core/tabletserver/log/LogEntry.java [hunk 2]
```diff
@@ -19,67 +19,124 @@
...
+      throw new IllegalArgumentException("Expected valid UUID. Found '" + uuidPart + "'");
+    }
+  }
+
+  /**
+   * Make a copy of this LogEntry but replace the file path.
+   *
+   * @param filePath path to use
+   */
+  public LogEntry switchFile(String filePath) {
```
[the remark this answers is not in the document]
Comment (keith-turner) [reply core/src/main/java/org/apache/accumulo/core/tabletserver/log/LogEntry.java:89]: Opened #3922

File: core/src/main/java/org/apache/accumulo/core/tabletserver/log/LogEntry.java [hunk 3]
```diff
@@ -19,67 +19,124 @@
...
+   * (host:port) followed by a UUID as the file name. For example,
+   * localhost:1234/927ba659-d109-4bce-b0a5-bcbbcb9942a2 is a valid file path.
+   *
+   * @param filePath path to validate
+   * @throws IllegalArgumentException if the filepath is invalid
+   */
+  private static void validateFilePath(String filePath) {
+    String[] parts = filePath.split("/");
+
+    if (parts.length < 2) {
```
[the remark this answers is not in the document]
Comment (keith-turner) [reply core/src/main/java/org/apache/accumulo/core/tabletserver/log/LogEntry.java:61]: If we remove the value, could make the validation even more strict.

[the remark this answers is not in the document]
Comment (DomGarguilo) [reply core/src/main/java/org/apache/accumulo/core/tabletserver/log/LogEntry.java:61]: So should paths with nested directories (e.g. `dir1/tserver:8020/UUID`) not be allowed?

[the remark this answers is not in the document]
Comment (keith-turner) [reply core/src/main/java/org/apache/accumulo/core/tabletserver/log/LogEntry.java:61]: > So should paths with nested directories (e.g. dir1/tserver:8020/UUID) not be allowed?

Nevermind, ignore this comment.  I was wrong about this.  I was assuming that full path were not being stored base on a comment in the class that I misunderstood. I added code to print what is stored and ran an IT that does log recovery and it is a full path.

File: core/src/main/java/org/apache/accumulo/core/tabletserver/log/LogEntry.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (keith-turner) [reply core/src/main/java/org/apache/accumulo/core/tabletserver/log/LogEntry.java:89]: This comment is wrong, its a full path and the method is still needed.

**Review by keith-turner — APPROVED** [review]

**Pull request merged**